### DISCUSSION TITLE
PID Tuning and a maibe stupid G28 question :-)

### DISCUSSION THREAD

**Discussion by Team-Deepflights** [phil-barrett/grblHAL-teensy-4.x#61] [General]
Hi,

I finally finished the lathe and it works as it should.

Only threading still makes me mad, the angles are not as they should be and thats why I reflashed the Teensy with PID logging (1000) enabled.
But if I click on the button, and than start a few thread cycles in the air, I see nothing at all on the PID logging.
The lathe have no VFD, I only can change RPM by turning a potentiometer. The RPM are measured with a sensor and the RPM are shown correctly in IOSender.
Could please someone explain me how the logs are visualized? Is there anything more to set up in the program?
How is the workflow to get the data?

The other issue is, I tried to write a macro with homing and set G92 to get a fixed zero to measute the tool offsets.
I used G28, did not know that $H is the way to go.
I am not sure, but I think since then if I look at the offsets I see x-1z-1 at the G28 row.
The tools had an wrong offset too. I corrected them but it seems to "forget" it, just right now I had this issue again, I wanted 12mm diameter and got 13mm.
How can I disable the G28?

Btw, it found a few "bugs" in IOSender 2.0.37:
I tried C&P the X Z values into the tooloffset, the values are there but if I click the button to save them, nothing happens. I had to retype them manually.
If a tool with offsets is loaded, the value is gone in the offset page if you click on "Reset", it´s lost at all. also rebootig keeps the affected toolfield to 0,00 0,00. ( I was lucky, I copied the values into a TXT) :-)

Thanks in advance.

**Comment by phil-barrett** [comment]
You are well past my level of understanding.  Perhaps you should ask your question in the [grblHAL core discussions](https://github.com/grblHAL/core/discussions) section.  I know Terje has posted several videos of threading, perhaps he can help you?

ioSender bugs should be reported [here](https://github.com/terjeio/ioSender/issues).

Phil

**Comment by Team-Deepflights** [comment]
Thanks Phil, I´ll copy the questions to the links above.
